src/Renderer.php

```diff
@@ -214,20 +302,58 @@ public function inlineJson($values)
      */
     public function processData($data)
     {
-        $data['themePrefix'] = $this->theme ? $this->theme . "::" : "";
+        $themePrefix = $this->theme ? $this->theme . "::" : "";
+
         if (isset($data['stripes'])) {
-            $data['stripes'] = array_map(function ($stripe) {
+            // filter out stripes that don't exist for the theme
+            $filteredStripes = array_filter(
+                $data['stripes'],
+                fn($s) => $this->engine->exists($themePrefix . 'stripes/' . $s['type'])
+            );
+
+            // get an array of indexes
+            $indexes = array_keys($filteredStripes);
+
+            // process each stripe when any defined `processStripe[type]` functions
+            $processedStripes = array_map(function ($stripe, $index) {
                 $hook = 'processStripe' . ucwords(preg_replace("/^stripe\-/", '', $stripe['type']));
-                return is_callable([$this, $hook])
-                    ? $this->$hook($stripe)
-                    : $stripe;
-            }, array_filter($data['stripes'], function ($stripe) use ($data) {
-                return $this->engine->exists($data['themePrefix'] . 'stripes/' . $stripe['type']);
-            }));
+                return is_callable([$this, $hook]) ? $this->$hook($stripe, $index) : $stripe;
+            }, $filteredStripes, $indexes);
+
+            $data['stripes'] = $processedStripes;
         }
+
+        $data['themePrefix'] = $themePrefix;
+
         return $data;
     }
 
+    /**
+     * Process stripe-textblock data
+     *
+     * @param  array $stripe array of data
+     * @return array
+     */
+    public function processStripeTextblock($stripe, $stripeIndex)
+    {
+        if (!empty($stripe['content']['html'])) {
+            // add a unique id to each h1 and h2 tag so they can be anchored to
```

Comment:
I think we talked about this, but what if we did this coming out of the API instead of here??? Same code probably, if this works well, but just on retrieval so we could in theory use anchors wherever we display this stripe.

Reply:
I don't remember but yeah that sounds better! Where would that live in the api code?

Reply:
Hm...maybe another case in `getFieldsAttribute()` on the stripe model?

Reply:
Ok cool I did that here: 
https://bitbucket.org/journeygroup/api.sites.provost.virginia.edu/pull-requests/6